dask/distributed/tests/test_ipython_utils.py

```diff
@@ -0,0 +1,68 @@
+import time
+from subprocess import Popen, PIPE
+
+import pytest
+
+np = pytest.importorskip('numpy')
+pytest.importorskip('IPython.parallel')
+
+import numpy as np
+from numpy.testing import assert_array_almost_equal
+
+from IPython.parallel import Client
+
+from dask.distributed import dask_client_from_ipclient
+import dask.array as da
+
+
+def setup_module():
+    start_cmd = ['ipcluster', 'start', '-n', '4']
+    cp = Popen(start_cmd, stdout=PIPE, stderr=PIPE)
+    tic = time.time()
+    while True:
+        line = cp.stderr.readline().decode()
+        if 'Engines appear to have started' in line:
+            break
+        elif 'CRITICAL | Cluster is already running' in line:
+            raise RuntimeWarning("Cluster is already running")
+        elif time.time() - tic > 50:
+            raise RuntimeError("Timeout waiting for cluster to start.")
+        time.sleep(0.1)
+
+
+def teardown_module():
+    kill_cmd = ['ipcluster', 'stop']
+    cp = Popen(kill_cmd, stdout=PIPE, stderr=PIPE)
+    tic = time.time()
+    while True:
+        line = cp.stderr.readline().decode()
+        if "Stopping cluster" in line:
+            break
+        elif time.time()-tic > 15:
+            # probably stopped or something went wrong
+            break
+        time.sleep(0.1)
```

Comment:
Just chiming in: I wouldn't really recommend importing our test case. It's tied to too much IPython stuff, and not really designed to be re-used. This might prompt me to clean it up and make it a reusable API, but for now, I'd copy what's useful and leave the rest.